RedHatInsights/insights-results-aggregator-cleaner
# [CCXDEV-12954] Rename dvo_report into dvo.dvo_report

# Description

Rename `dvo_report` into `dvo.dvo_report`.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Testing steps

Locally:

1. Run `podman-compose up -d`
2. Check the dvo-writer created the table:
```
aggregator=# SELECT * FROM dvo.dvo_report;
 org_id | cluster_id | namespace_id | namespace_name | report | recommendations | objects | reported_at | last_checked_at | rule_hits_count 
--------+------------+--------------+----------------+--------+-----------------+---------+-------------+-----------------+-----------------
(0 rows)
```
3. Fill the DB with some mock data:
```
❯ ./insights-results-aggregator-cleaner -fill-in-db
{"level":"info","filename":"config","time":"2024-06-26T09:29:54+02:00","message":"parsing configuration file"}
Clowder is disabled
9:29AM DBG Started
9:29AM INF DB connection configuration driverName=postgres
9:29AM INF Fill-in database started
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF inserting into DVO database Insert statement="\n\t    INSERT INTO dvo.dvo_report\n\t           (org_id, cluster_id, namespace_id, namespace_name, report, recommendations, objects, reported_at, last_checked_at, rule_hits_count)\n\t\t   values\n\t\t   ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);"
9:29AM INF Fill-in DVO database finished
9:29AM DBG Finished
```
4. Check with a dry-run:
```
❯ ./insights-results-aggregator-cleaner -cleanup-all -dry-run
{"level":"info","filename":"config","time":"2024-06-26T09:30:01+02:00","message":"parsing configuration file"}
Clowder is disabled
9:30AM DBG Started
9:30AM INF DB connection configuration driverName=postgres
9:30AM DBG Cleaning all old records from DB Max age="90 days"
9:30AM INF Cleanup-all started
9:30AM INF Delete records Affected=6 Dry run=true table=dvo.dvo_report
9:30AM INF Cleanup-all finished
9:30AM DBG Finished
```
5. Check in psql
```
aggregator=# SELECT * FROM dvo.dvo_report;
 org_id |              cluster_id              |             namespace_id             | namespace_name | report | recommendations | objects |     reported_at     |   last_checked_at   | rule_hits_count 
--------+--------------------------------------+--------------------------------------+----------------+--------+-----------------+---------+---------------------+---------------------+-----------------
      1 | 00000001-0001-0001-0001-000000000001 | fbcbe2d3-e398-4b40-9d5e-4eb46fe8286f | not set        |        |               1 |       6 | 2021-01-01 00:00:00 | 2021-01-01 00:00:00 | {}
      1 | 00000002-0002-0002-0002-000000000002 | e6ed9bb3-efc3-46a6-b3ae-3f1a6e59546c | not set        |        |               2 |       5 | 2021-01-01 00:00:00 | 2021-01-01 00:00:00 | {}
      2 | 00000003-0003-0003-0003-000000000003 | e6ed9bb3-efc3-46a6-b3ae-3f1a6e59546c | not set        |        |               3 |       4 | 2021-01-01 00:00:00 | 2021-01-01 00:00:00 | {}
      3 | 00000001-0001-0001-0001-000000000001 | e6ed9bb3-efc3-46a6-b3ae-3f1a6e59546c | not set        |        |               4 |       3 | 2021-01-01 00:00:00 | 2021-01-01 00:00:00 | {}
      3 | 00000002-0002-0002-0002-000000000002 | e6ed9bb3-efc3-46a6-b3ae-3f1a6e59546c | not set        |        |               5 |       2 | 2022-01-01 00:00:00 | 2022-01-01 00:00:00 | {}
      3 | 00000003-0003-0003-0003-000000000003 | e6ed9bb3-efc3-46a6-b3ae-3f1a6e59546c | not set        |        |               6 |       1 | 2023-01-01 00:00:00 | 2023-01-01 00:00:00 | {}
(6 rows)
```

6. Clean up
```
❯ ./insights-results-aggregator-cleaner -cleanup-all -dry-run=false
{"level":"info","filename":"config","time":"2024-06-26T09:30:09+02:00","message":"parsing configuration file"}
Clowder is disabled
9:30AM DBG Started
9:30AM INF DB connection configuration driverName=postgres
9:30AM DBG Cleaning all old records from DB Max age="90 days"
9:30AM INF Cleanup-all started
9:30AM INF Delete records Affected=6 Dry run=false table=dvo.dvo_report
9:30AM INF Cleanup-all finished
9:30AM DBG Finished
```

7. Check in psql
```
aggregator=# SELECT * FROM dvo.dvo_report;
 org_id | cluster_id | namespace_id | namespace_name | report | recommendations | objects | reported_at | last_checked_at | rule_hits_count 
--------+------------+--------------+----------------+--------+-----------------+---------+-------------+-----------------+-----------------
(0 rows)
```

## Checklist
